mistlib/core.py

```diff
@@ -356,7 +356,145 @@ def append_file(input_filename, output_filename):
+            first_value = pair[0]
+            labeled_values[label] = first_value
+        thermal_cond_solid = []
+        for label, value in labeled_values.items():
+            thermal_cond_solid.append(value)
+            print(f"{value}")
+
+        p=(self.properties["thermal_conductivity_liquid"].value_laurent_poly)
+        first_values = [pair[0] for pair in p]
+        labeled_values = {}
+        for idx, pair in enumerate(p, 1):
+            label = f"thermal_cond_L{idx}" 
+            first_value = pair[0]
+            labeled_values[label] = first_value
+        thermal_cond_liquid = []
+        for label, value in labeled_values.items():
+            thermal_cond_liquid.append(value)
+            print(f"{value}")
+
+        p = (self.properties["specific_heat_solid"].value_laurent_poly)
+        first_values = [pair[0] for pair in p]
+        labeled_values = {}
+        for idx, pair in enumerate(p, 1):
+            label = f"specific_heat_S{idx}" 
+            first_value = pair[0]
+            labeled_values[label] = first_value
+        specific_heat_solid = []
+        for label, value in labeled_values.items():
+            specific_heat_solid.append(value)
+            print(f"{value}")
+
+        p = (self.properties["specific_heat_liquid"].value_laurent_poly)
+        if self.properties["specific_heat_liquid"] == ValueTypes.LAURENT_POLYNOMIAL:
+            first_values = [pair[0] for pair in p]
+            labeled_values = {}
+            for idx, pair in enumerate(p, 1):
+                label = f"specific_heat_L{idx}" 
+                first_value = pair[0]
+                labeled_values[label] = first_value
+            specific_heat_liquid= []
+            for label, value in labeled_values.items():
+                specific_heat_liquid.append(value)
+                print(f"{value}")
+        else:
+            p = self.properties["specific_heat_liquid"].value
+            specific_heat_liquid = []
+            specific_heat_liquid.append(self.properties["specific_heat_liquid"].value) 
+            print(f"{value}")
+        
+        density = self.get_property("density", code_name, reference_temperature)
+        latent_heat_fusion = self.get_property("latent_heat_fusion", code_name, reference_temperature)
+        dynamic_viscosity = self.get_property("dynamic_viscosity", code_name, reference_temperature)
+        thermal_expansion = self.get_property("thermal_expansion", code_name, reference_temperature)
+        
+        with open(file, "w") as f:
+            f.write(comment_block)
+            f.write("solid\n{\n")
+            
+            p = self.properties["thermal_conductivity_solid"]
+            if (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
+                f.write(f"\tkappa\t ({thermal_cond_solid[0]} {thermal_cond_solid[1]} 0.0);\n")
+            else:
+                f.write(f"\tkappa\t ({self.properties['thermal_conductivity_solid'].value} 0.0 0.0);\n")
```

Comment:
Looks like another good helper function (additivefoam specific this time though)

Reply:
Added